File: 卡码/代码随想录/Python/贪心/01-简单/LeetCode_1005.py
#1005. K 次取反后最大化的数组和
from typing import *

class Solution:
  def largestSumAfterKNegations(self, nums: List[int], k: int) -> int:
    nums.sort(key=lambda x:abs(x),reverse=True)
    res:int=0
    for i in range(len(nums)):
      if k>0 and nums[i]<0:
        nums[i]=-nums[i]
        k-=1
      res+=nums[i]
    if k%2==1:
      res-=2*nums[len(nums)-1]
    return res

File: 卡码/代码随想录/C++/贪心/01-简单/LeetCode_1005.cpp
//1005. K 次取反后最大化的数组和
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
	int largestSumAfterKNegations(vector<int>& nums, int k) {
		sort(nums.begin(),nums.end(),[](int a,int b){return abs(a)>abs(b);});
		for(int i=0;i<nums.size();i++){
			if(nums[i]<0&&k>0){
				k--;
				nums[i]=-nums[i];
			}
		}
		if(k%2==1) nums[nums.size()-1]*=-1;
		return accumulate(nums.begin(),nums.end(),0);
	}
};

int main(){
  system("pause");
  return 0;
}
